superset/views/utils.py

```diff
@@ -184,3 +184,10 @@ def get_datasource_info(datasource_id, datasource_type, form_data):
             'The datasource associated with this chart no longer exists')
     datasource_id = int(datasource_id)
     return datasource_id, datasource_type
+
+def apply_display_max_row_limit(data):
```

Comment:
NIT: `data` is vague here, maybe `sql_results` (given it's the complex data structure coming out of `get_sql_results`)

Reply:
Maybe add a docstring explaining why it's mutating this `sql_results` payload, something like:
```
"""
Given a `sql_results` nested structure, applies a limit to the number of rows

`sql_results` here is the nested structure coming out of sql_lab.get_sql_results, it contains
metadata about the query, as well as the data set returned by the query. This method
limits the number of rows adds a `displayLimitReached: True` flag to the metadata
"""
```